libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1074,64 +1074,91 @@ internal object MapboxRouteLineUtils {
         return expressionBuilder.build()
     }
 
-    private fun addSource(
-        style: Style,
-        layerSource: String,
+    /**
+     * Compares provided [properties] map with existing values for a source with [sourceId].
+     *
+     * The primary mechanism to compare the values is [Value.equals],
+     * however, there's a special case [Double]s which are compared with a delta.
+     * Delta comparison is not supported for collections
+     * but it's a case that this class currently doesn't need to take into account.
+     */
+    private fun Style.sourcePropertiesCompatible(
+        sourceId: String,
+        properties: HashMap<String, Value>
+    ) = properties.all {
+        val existing = getStyleSourceProperty(sourceId, it.key).value
+        val existingContents = existing.contents
+        val expected = it.value
+        val expectedContents = expected.contents
+        if (existingContents is Double && expectedContents is Double) {
```

Comment:
Is Float also possible here?

Reply:
No, source is [here](https://github.com/mapbox/mapbox-maps-android/blob/b69460ef22e089fbdf4914d40631c57e149e610a/extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt#L112-L143) and [here](https://github.com/mapbox/mapbox-maps-android/blob/b69460ef22e089fbdf4914d40631c57e149e610a/extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt#L50-L53).

That said, I can add an overload so that we don't depend on impl details of the Maps SDK.